File: backend/Application/DTOs/AllRequests/Requests/Carts/AddToCartDto.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.DTOs.AllRequests.Requests.Carts
{
    public class AddToCartDto
    {
        public int DonationId { get; set; }
        public string ItemName { get; set; }
        public int Quantity { get; set; }
        public CartType CartType { get; set; }
    }
    public class CartItemRequestDto
    {
        public int DonationId { get; set; }
        public string ItemName { get; set; }
        public int Quantity { get; set; }
        public CartType CartType { get; set; }
    }
    public class CartItemResponseDto
    {
        public List<CartItemRequestDto> Equipment { get; set; } = new();
        public List<CartItemRequestDto> Medicine { get; set; } = new();
    }
}

File: frontend/lib/features/auth/data/Cart/add_to_cart_dto.dart
class AddToCartDto {
  final int donationId;
  final String itemName;
  final int quantity;
  final int cartType; // 1 = Equipment, 2 = Medicine

  AddToCartDto({
    required this.donationId,
    required this.itemName,
    required this.quantity,
    required this.cartType,
  });

  Map<String, dynamic> toJson() {
    return {
      "donationId": donationId,
      "itemName": itemName,
      "quantity": quantity,
      "cartType": cartType,
    };
  }
}
